src/js/components/file-input.js

```diff
@@ -225,7 +225,8 @@ const preventInvalidFiles = (e, fileInputEl, instructions, dropTarget) => {
       removeOldPreviews(dropTarget, instructions);
       fileInputEl.value = ""; // eslint-disable-line no-param-reassign
       dropTarget.insertBefore(errorMessage, fileInputEl);
-      errorMessage.innerHTML = `This is not a valid file type.`;
+      errorMessage.innerHTML =
+        fileInputEl.dataset.errormessage || `This is not a valid file type.`
```

Comment:
Depending whether the `data-` attributes could be considered untrusted (e.g. user-contributed), maybe this ought to switch from `innerHTML` to `textContent`, to avoid any markup in the attribute value being executed?

```suggestion
      errorMessage.textContent =
        fileInputEl.dataset.errormessage || `This is not a valid file type.`
```

References:

- https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#differences_from_innerhtml
- https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html#rule-6-populate-the-dom-using-safe-javascript-functions-or-properties

Reply:
Good idea, added.